pkg/build/controller/config_controller.go

```diff
@@ -33,11 +37,32 @@ func IsFatal(err error) bool {
 
 type BuildConfigController struct {
 	BuildConfigInstantiator buildclient.BuildConfigInstantiator
+
+	KubeClient kclient.Interface
+	Client     osclient.Interface
 }
 
 func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) error {
 	glog.V(4).Infof("Handling BuildConfig %s/%s", bc.Namespace, bc.Name)
 
+	if strategy := bc.Spec.Strategy.JenkinsPipelineStrategy; strategy != nil {
+		glog.V(4).Infof("Detected Jenkins pipeline strategy in %s/%s build configuration", bc.Namespace, bc.Name)
+		if _, err := c.KubeClient.Services(bc.Namespace).Get(buildapi.JenkinsDefaultServiceName); err == nil {
```

Comment:
the config controller is wrapped by a retry controller (like most of our controllers):
https://github.com/openshift/origin/blob/master/pkg/build/controller/factory/factory.go#L315-L335

so it all depends what HandleBuildConfig returns as an error, as to whether the event gets retried or not.  If you return a ConfigControllerFatalError, it won't.  otherwise it will.

it's hard to imagine a situation where we should not retry this since we should always expect to be able to reach the api server.  Note that the retry is limited, so eventually it does give up, but presumably the reget/list flow should start the retry again when that happens.